src/builtins/exec.rs

```diff
@@ -0,0 +1,47 @@
+use crate::builtins::InternalCommandContext;
+use crate::process::ExitStatus;
+use nix::errno::Errno;
+use std::ffi::CString;
+use std::io::Write;
+
+pub fn command(ctx: &mut InternalCommandContext) -> ExitStatus {
+    if let Some(command) = ctx.argv.get(1) {
+        let command = match CString::new(command.as_bytes()) {
+            Ok(args) => args,
+            Err(_) => {
+                writeln!(ctx.stderr, "nsh: exec: nul found in inputs").ok();
```

Comment:
I'd make this error message more specific:

```suggestion
                writeln!(ctx.stderr, "nsh: exec: invalid command (perhaps it includes a NUL character?)").ok();
```